examples/gno.land/p/demo/microblog/microblog.gno

```diff
@@ -0,0 +1,174 @@
+package microblog
+
+import (
+	"errors"
+	"std"
+	"strings"
+	"time"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ufmt"
+)
+
+type Microblog struct {
+	Title  string
+	Prefix string   // i.e. r/gnoland/blog:
+	Pages  avl.Tree // author (string) -> Page
+}
+
+func NewMicroblog(title string, prefix string) (m *Microblog) {
+	return &Microblog{
+		Title:  title,
+		Prefix: prefix,
+		Pages:  avl.Tree{},
+	}
+}
+
+func (m *Microblog) GetPages() []*Page {
+	var (
+		pages = make([]*Page, m.Pages.Size())
+		index = 0
+	)
+
+	m.Pages.Iterate("", "", func(page *avl.Node) bool {
+		pages[index] = page.Value().(*Page)
+		index++
+		return false
+	})
+
+	// TODO: sort by last updated
+
+	return pages
+}
+
+func (m *Microblog) Render(path string) string {
+	parts := strings.Split(path, "/")
+
+	isHome := path == ""
+	isUser := len(parts) == 1
+
+	switch {
+	case isHome:
+		output := ufmt.Sprintf("# %s\n\n", m.Title)
+		output += "# pages\n\n"
+
+		for _, page := range m.GetPages() {
+			output += ufmt.Sprintf("- [%s](%s%s)\n", page.Author.String(), m.Prefix, page.Author.String())
+		}
+
+		return output
```

Comment:
You can probably extract this out into a smaller helper method for better readability

Reply:
Great idea, done: https://github.com/schollz/gno/commit/3c05a6779ed2d1559b2785c6b664db0dc331ed7b